src/plugins/discourse/mirror.test.js

```diff
@@ -568,4 +654,548 @@ describe("plugins/discourse/mirror", () => {
+      fetcher.addPost(3, null, "beta");
+      const l1 = fetcher.addLike("beta", 1, 5);
+      const l2 = fetcher.addLike("beta", 2, 6);
+      const l3 = fetcher.addLike("beta", 3, 7);
+      const l4 = fetcher.addLike("alpha", 1, 8);
+      await mirror.update(reporter);
+      expectLikesSorted(repo.likes(), [l1, l2, l3, l4]);
+
+      const l5 = fetcher.addLike("alpha", 2, 9);
+      fetcher.addPost(4, null, "credbot");
+      const l6 = fetcher.addLike("credbot", 2, 10);
+      const l7 = fetcher.addLike("beta", 4, 11);
+      await mirror.update(reporter);
+      expectLikesSorted(repo.likes(), [l1, l2, l3, l4, l5, l6, l7]);
+    });
+
+    it("doesn't find likes of users that never posted", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+      fetcher.addPost(1, null);
+      fetcher.addLike("nope", 1, 1);
+      await mirror.update(reporter);
+      expect(repo.likes()).toEqual([]);
+    });
+
+    it("mirrors category definition topics only after it's interval", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+
+      fetcher.addCategory(4);
+      const topic4 = fetcher._categoryDefinitionTopic(4);
+      await mirror.update(reporter);
+      const firstTopics = repo.topics();
+
+      fetcher.addCategory(5);
+      await mirror.update(reporter);
+      const laterTopics = repo.topics();
+
+      expect(firstTopics).toEqual([topic4]);
+      expect(laterTopics).toEqual([topic4]);
+    });
+
+    it("mirrors respects recheckCategoryDefinitionsAfterMs option", async () => {
+      const {mirror, fetcher, reporter, repo} = example({
+        recheckCategoryDefinitionsAfterMs: 0,
+      });
+
+      fetcher.addCategory(4);
+      const topic4 = fetcher._categoryDefinitionTopic(4);
+      await mirror.update(reporter);
+      const firstTopics = repo.topics();
+
+      fetcher.addCategory(5);
+      const topic5 = fetcher._categoryDefinitionTopic(5);
+      await mirror.update(reporter);
+      const laterTopics = repo.topics();
+
+      expect(firstTopics).toEqual([topic4]);
+      expect(laterTopics).toEqual([topic4, topic5]);
+    });
+
+    it("mirrors respects recheckTopicsInCategories option", async () => {
+      const {mirror, fetcher, reporter, repo} = example({
+        recheckTopicsInCategories: [4],
+      });
+      const cooked = "<p>New content, without bump</p>";
+
+      // Adds a category description #4.
+      // And a topic 5 which is a member of the #4 category.
+      const p1 = fetcher.addPost(1, null);
+      fetcher.addPost(2, null);
+      fetcher.addPost(3, null);
+      fetcher.addCategory(4);
+      const p5 = fetcher.addPost(5, null);
+      fetcher.setTopicCategory(5, 4);
+
+      // See we're trying to update #4 and #5.
+      const topicsCall = jest.spyOn(fetcher, "topicWithPosts");
+      await mirror.update(reporter);
+
+      fetcher.editPost(p1, {cooked});
+      fetcher.editPost(p5, {cooked});
+      const earlyCalls = [...topicsCall.mock.calls];
```

Comment:
(personally, I would use topicsCalls.mock.calls.slice(), but this is fine too)

Reply:
I find spreading into a new array literal a bit more obvious to be used as a shallow copy than `.slice()`. Especially because array slices are a concept in other languages that don't make shallow copies. Such as Go.

https://blog.golang.org/go-slices-usage-and-internals#TOC_4.
>  A slice is a descriptor of an array segment. It consists of a pointer to the array, the length of the segment, and its capacity (the maximum length of the segment).

Reply:
makes sense.